conf/flight_plans/rotorcraft_guided_flightplan.xml

```diff
@@ -0,0 +1,105 @@
+<!DOCTYPE flight_plan SYSTEM "flight_plan.dtd">
+
+<flight_plan alt="1.0" ground_alt="0" lat0="51.990634" lon0="4.376789" max_dist_from_home="20" name="Rotorcraft Optitrack (Delft)" security_height="0.3">
+  <header>
+
+#include "autopilot.h"
+
+/********************************************//**
+ *  ABI messages for exception triggers or added values for the guided flight plan
+ ***********************************************/
+
+// ABI for obstacle detection (In here it comes from the module colorfilter)
+#include "subsystems/abi.h"
+
+#ifndef FP_GUIDED_OBSTACLE_DETECTION_ID
+#define FP_GUIDED_OBSTACLE_DETECTION_ID ABI_BROADCAST    
+PRINT_CONFIG_VAR(FP_GUIDED_OBSTACLE_DETECTION_ID)
+#endif
+
+static abi_event obstacle_detection_ev;
+static void obstacle_detection_cb(uint8_t sender_id, float distance_obstacle, float heading_obstacle);
+float distance_obstacle_FP;
+static void obstacle_detection_cb(uint8_t UNUSED(sender_id), float distance_obstacle, float UNUSED(heading_obstacle))
+{
+  distance_obstacle_FP = distance_obstacle;
+}
```

Comment:
By doing it like this, it is creating static variables everywhere the flight_plan.h header is included. It may work, but I don't think it is what we want. You could at least do something like this:
```
#ifdef NAV_C
abi_event obstacle_detection_ev;
float distance_obstacle_FP = 0.f;
void obstacle_detection_cb(uint8_t UNUSED(sender_id), float distance_obstacle, float UNUSED(heading_obstacle))
{
  distance_obstacle_FP = distance_obstacle;
}
#else
extern abi_event obstacle_detection_ev;
extern float distance_obstacle_FP;
extern void obstacle_detection_cb(uint8_t sender_id, float distance_obstacle, float heading_obstacle);
#endif
```
I guess the proper solution is to generate this code automatically from a new dedication xml description in the flight plan. I'll think about it...